discord/ext/menus/__init__.py

```diff
@@ -1062,13 +1157,23 @@ async def get_page(self, page_number):
         """
         if self.per_page == 1:
             return self.entries[page_number]
-        else:
-            base = page_number * self.per_page
-            return self.entries[base:base + self.per_page]
+        base = page_number * self.per_page
+        return self.entries[base : base + self.per_page]
 
-_GroupByEntry = namedtuple('_GroupByEntry', 'key items')
 
-class GroupByPageSource(ListPageSource):
+class _GroupByEntry(tuple, Generic[KT, IT]):
```

Comment:
When subclassing GroupByPageSource.format_page it is going to be strange to have to use something which appears to be a private class to ext.menus to use type hints. I think this should be renamed to ```GroupByEntry`` to avoid this.